subsys/bluetooth/host/smp.c

```diff
@@ -2908,6 +2945,45 @@ static u8_t sc_smp_check_confirm(struct bt_smp *smp)
 	return 0;
 }
 
+static bool le_sc_oob_data_req_check(struct bt_smp *smp)
+{
+	struct bt_smp_pairing *req = (struct bt_smp_pairing *)&smp->preq[1];
+
+	return ((req->oob_flag & BT_SMP_OOB_DATA_MASK) == BT_SMP_OOB_PRESENT) ?
+		true : false;
```

Comment:
`(a == b)` is already in itself a valid boolean statement (something you can assign to a `bool` type variable), so the extra `(a == b) ? true : false` seems unnecessary.